### PR TITLE
Add router and top-level application class

### PR DESCRIPTION
**Pull request by thisandagain**
- Adds a stub class called `Router` to allow for URL-style activity routing
- Adds a top-level application class called `WebmakerApplication` which allows for:
    - Router setup
    - Google Analytics (todo)
    - UserTesting.com SDK (todo)
    - Current activity state save / restoration after application exit (todo)

### CODE REVIEW

**Comment by thisandagain** [comment]
@ryanw-ny Here is a quick starting point. Thoughts?

File: app/src/main/java/mozilla/org/webmaker/MainActivity.java [hunk 1]
```diff
@@ -24,6 +25,8 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        Log.v("[Webmaker]", "MainActivity created.");
```
Comment (k88hudson) [inline app/src/main/java/mozilla/org/webmaker/MainActivity.java]: intentional?

File: package.json [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
   "scripts": {
     "start": "npm run build && npm-run-all --parallel webserver watch:static watch:js watch:css watch:html",
     "build:clean": "rimraf app/src/main/assets/www/ mkdirp app/src/main/assets/www/",
-    "build:static": "mkdirp www_src/static/ ncp www_src/static/ app/src/main/assets/www/",
+    "build:static": "mkdirp www_src/static/ && mkdirp app/src/main/assets/ && ncp www_src/static/ app/src/main/assets/www/",
     "build:js": "webpack",
     "build:css": "npm run watch:css -- --no-watch",
     "build:html": "node ./npm_tasks/build-html",
-    "build": " npm-run-all build:clean build:static build:js build:css",
+    "build": " npm-run-all build:clean build:static build:js build:css build:html",
```
Comment (k88hudson) [inline package.json]: good catch

File: app/src/main/java/mozilla/org/webmaker/MainActivity.java (hunk 1 above)
Comment (thisandagain) [inline app/src/main/java/mozilla/org/webmaker/MainActivity.java]: Yeah... I like having lifecycle events in the ADB logs. Happy to remove these types of logs if you prefer.

**Comment by thisandagain** [comment]
@k88hudson R? [[flagged!]](http://plzkthx.herokuapp.com)

**Comment by k88hudson** [comment]
Looks good to me. Let's talk about how we can pass stuff from the router context to the webview

**Pull request merged**